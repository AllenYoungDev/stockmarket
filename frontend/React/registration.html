<html>
<head>
<style>
.top-bar { margin: 0; padding: 1% 0.5% 1% 0; background: rgb(0,0,0); color: rgb(255,255,255); text-align: left; font-family: Arial, Helvetica, sans-serif; }

@media only screen and (orientation: landscape) {
.registration-form-section {margin: 0;background-color: #737CA1;color: rgb(250,250,250);}
.registration-form-section-heading {text-align: center; font-size: 250%; font-family: Arial, Helvetica, sans-serif; font-weight: 520; margin: 0;}
.registration-form {margin: 1% 0% 0% 0%;}
.registration-form-div-grid {display:grid; grid-template-columns: 33% 12% 12% 33%; grid-gap:5px; font-size: 130%;font-family: arial;}
.registration-form-label-first-name {grid-row: 1; grid-column: 2; text-align: right;}
.registration-form-input-first-name {grid-row: 1; grid-column: 3; height: 2em; }
.registration-form-label-last-name {grid-row: 2; grid-column: 2;text-align: right;}
.registration-form-input-last-name {grid-row: 2; grid-column: 3; height: 2em; }
.registration-form-div-below-first-last-names {margin: 2.5%;grid-row: 3; grid-column: 2;}
.registration-form-label-email-address {grid-row: 4; grid-column: 2;text-align: right;}
.registration-form-input-email-address {grid-row: 4; grid-column: 3; grid-gap:15px; height: 2em; }
.registration-form-label-phone-number {grid-row: 5; grid-column: 2;text-align: right;}
.registration-form-input-phone-number {grid-row: 5; grid-column: 3; height: 2em; }
.registration-form-div-below-phone-number {margin: 2.5%;grid-row: 6; grid-column: 2;}
.registration-form-label-password {grid-row: 7; grid-column: 2;text-align: right;}
.registration-form-input-password {grid-row: 7; grid-column: 3; height: 2em; }
.registration-form-password-show-hide {grid-row: 7; grid-column: 4;margin: 0; padding: 0.35em 0 0 0.25em;text-align: left;font-size: 60%;font-family: Arial, Helvetica, sans-serif; font-weight: 50;}
.registration-form-label-password-retype {grid-row: 8; grid-column: 2;text-align: right;}
.registration-form-input-password-retype {grid-row: 8; grid-column: 3; height: 2em; }
.registration-form-paragraph-notice1 {text-align: center; margin: 1% 0 0 0;font-size: 120%;}
.registration-form-paragraph-notice2 {text-align: center; margin: 0.5% 0 0 0;font-size: 120%;}
.registration-form-paragraph-notice3 {text-align: center; margin: 0.5% 0 0 0;font-size: 120%;}
.registration-form-div-register-button {text-align: center;}
.registration-form-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}
.registration-form-input-register-button {margin: 1% 0 1.25% 0; padding: 0.5%; font-size: 135%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250);}
}
@media only screen and (orientation: portrait) {
.registration-form-section {margin: 0;background-color: #737CA1;color: rgb(250,250,250);}
.registration-form-section-heading {text-align: center; font-size: 250%; font-family: Arial, Helvetica, sans-serif; font-weight: 520; margin: 0 0 3% 0;}
.registration-form {margin: 1% 0% 0% 0%; font-size: 200%;}
.registration-form-div-grid {display:grid; grid-template-columns: 40% 50% 10%; font-size: 130%;font-family: arial;}
.registration-form-label-first-name {grid-row: 1; grid-column: 1; text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-first-name {grid-row: 1; grid-column: 2; height: 1.2em; font-size: 130%; margin-bottom: 2%;}
.registration-form-label-last-name {grid-row: 2; grid-column: 1;text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-last-name {grid-row: 2; grid-column: 2; height: 1.2em; font-size: 130%;}
.registration-form-div-below-first-last-names {margin: 2.5%;grid-row: 3; grid-column: 1;}
.registration-form-label-email-address {grid-row: 4; grid-column: 1;text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-email-address {grid-row: 4; grid-column: 2; grid-gap:15px; height: 1.2em; font-size: 130%;margin-bottom: 2%;}
.registration-form-label-phone-number {grid-row: 5; grid-column: 1;text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-phone-number {grid-row: 5; grid-column: 2; height: 1.2em; font-size: 130%;}
.registration-form-div-below-phone-number {margin: 2.5%;grid-row: 6; grid-column: 1;}
.registration-form-label-password {grid-row: 7; grid-column: 1;text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-password {grid-row: 7; grid-column: 2; height: 1.2em; font-size: 130%;margin-bottom: 2%;}
.registration-form-password-show-hide {grid-row: 7; grid-column: 3;margin: 0; padding: 0.35em 0 0 0.25em;text-align: left;font-size: 60%;font-family: Arial, Helvetica, sans-serif; font-weight: 50;}
.registration-form-label-password-retype {grid-row: 8; grid-column: 1;text-align: right; margin-right: 3%;font-size: 130%;}
.registration-form-input-password-retype {grid-row: 8; grid-column: 2; height: 1.2em; font-size: 130%;}
.registration-form-paragraph-notice1 {text-align: center; margin: 3% 0 2% 0;font-size: 120%;}
.registration-form-paragraph-notice2 {text-align: center; margin: 0.5% 0 2% 0;font-size: 120%;}
.registration-form-paragraph-notice3 {text-align: center; margin: 0.5% 0 0 0;font-size: 120%;}
.registration-form-div-register-button {text-align: center;}
.registration-form-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}
.registration-form-input-register-button {margin: 2% 0 2.25% 0; padding: 0.5%; font-size: 200%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250);}
}

.footer   {margin: 0; padding: 0.25% 0.5% 0.1% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; text-align: center;}
</style>

</head>
<body style="margin: 0; background-color: #FFFFFF;color: rgb(250,250,250);">

<div id="user-registration-page"></div>




<!-- React component below, AccountRegistrationForm -->
<!--
<form class="settings" action="/action_page.php" style="margin: 1% 0% 0% 0%;">
  <div style="display:grid; grid-template-columns: 20% 20% 10% 20% 30%; grid-gap:5px; font-size: 130%;font-family: arial;">
  <label for="fname" style="grid-row: 1; grid-column: 1; text-align: right;">First name</label>
  <input type="text" id="fname" name="fname" size="20" style="grid-row: 1; grid-column: 2;">
  <label for="lname" style="grid-row: 1; grid-column: 3;text-align: right;">Last name</label>
  <input type="text" id="lname" name="lname" size="20" style="grid-row: 1; grid-column: 4;">
  <div style="margin: 2.5%;grid-row: 2; grid-column: 2;"></div>
  <label for="emailaddress" style="grid-row: 3; grid-column: 1;text-align: right;">Email address</label>
  <input type="text" id="emailaddress" name="emailaddress" size="20" style="grid-row: 3; grid-column: 2; grid-gap:15px;">  
  <label for="phonenumber" style="grid-row: 4; grid-column: 1;text-align: right;">Phone number</label>
  <input type="text" id="phonenumber" name="phonenumber" size="20" style="grid-row: 4; grid-column: 2;"> 
  <div style="margin: 2.5%;grid-row: 5; grid-column: 2;"></div>
  <label for="password" style="grid-row: 6; grid-column: 1;text-align: right;">Password</label>
  <input type="text" id="password" name="password" size="20" style="grid-row: 6; grid-column: 2;">
  <label for="passwordretype" style="grid-row: 7; grid-column: 1;text-align: right;">Password (Retype)</label>
  <input type="text" id="passwordretype" name="passwordretype" size="20" style="grid-row: 7; grid-column: 2;">   
  </div>

  <p style="text-align: center; margin: 1% 0 0 0;font-size: 120%;">Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~`!@#$%^&*-_+=|\/?;:"'<>,.{}[]().</p>
  <p style="text-align: center; margin: 0.5% 0 0 0;font-size: 120%;">Your email address will be your login ID.</p>
  <p style="text-align: center; margin: 0.5% 0 0 0;font-size: 120%;">After pressing the register button below, you'll receive an email-address verification email with a link that you must click to verify your email address.  You will not be able to log in before verifying your email address.</p>

  <div style="text-align: center;">
  <input type="submit" value="Register" style="margin: 1% 0 1.25% 0; padding: 0.5%; font-size: 150%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250);">
  </div>
</form>
-->


<script src="https://unpkg.com/react@18/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js" crossorigin></script>

<script type="module" src="registration.render.js"></script>

<noscript>Your browser must support JavaScript.</noscript>
</body>
</html>