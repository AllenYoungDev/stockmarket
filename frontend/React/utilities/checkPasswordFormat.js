const reLetterChecker = /[a-zA-Z]/;
const reNumberChecker = /[0-9]/;
const reSpecialCharacterChecker = /[\~\`\!\@\#\$\%\^\&\*\-\_\+\=\|\\\/\?\;\:\"\'\<\>\,\.\{\}\[\]\(\)]/;

export function checkPasswordFormat(password) {
    var numberOfCharactersInPassword = password.trim().length;

    if (numberOfCharactersInPassword < 10) {return false;}

    if (password.match(reLetterChecker) === null) {return false;}

    if (password.match(reNumberChecker) === null) {return false;}

    if (password.match(reSpecialCharacterChecker) === null) {return false;}

    return true;
}