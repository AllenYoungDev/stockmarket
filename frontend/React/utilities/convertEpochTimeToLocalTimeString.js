export function convertEpochTimeToLocalTimeString(epochTime) {
    var date = new Date(epochTime);
    return date.toLocaleDateString("default") + " " + date.toLocaleTimeString("default");
}