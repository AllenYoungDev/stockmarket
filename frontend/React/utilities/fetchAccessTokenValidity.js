'use strict';

import '../settings/ServerUrls.js';

export function fetchAccessTokenValidity(setLoginState, ignore) {

    console.log('fetchAccessTokenValidity() start.')

    fetch(window.backendServerUrl + "/CheckAccessTokenValidity", {method: "GET", credentials: 'include'})
    .then((response) => response.text())
    .then((text) => {
        if (text === 'valid') {
            console.log('fetchAccessTokenValidity():  access token is valid.');
            if (ignore) {return;}
            if (document.cookie.includes('admin=true')) {
                setLoginState('admin-login');
            } else {
                setLoginState('non-admin-login');
            }
        } else {
            console.log('fetchAccessTokenValidity():  access token is invalid.');
            if (ignore) {return;}
            setLoginState('no-login');
        }
    })
    .catch((err) => {
        console.log(`fetchAccessTokenValidity():  fetch() network error.  ${err}`);
        if (ignore) {return;}
        setLoginState('fetch-error');
    });
}