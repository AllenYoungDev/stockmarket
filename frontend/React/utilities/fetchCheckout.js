import '../settings/ServerUrls.js';

export function fetchCheckout(numberOfSharesToBuy, errorNoticeDomNodeRef) {
  let checkoutResponse = 0;
  let checkoutResponseStatus = 0;

  fetch(window.backendServerUrl + "/Checkout", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },      
    body: JSON.stringify({'numberOfSharesToBuy': numberOfSharesToBuy}),
    credentials: 'include',
    })
  .then((response) => {
    checkoutResponse = response;
    response.headers.forEach((value, key) => {
      console.log(`CheckoutButton() handleOnClick() checkoutResponse.headers key==>value:  ${key} ==> ${value}`)
    });
    checkoutResponseStatus=response.status;
    return response.text();})
  .then((text) => {
    if(checkoutResponseStatus === 200) {
      //window.location.href = window.frontendServerUrl;
      //This is not the proper implementation.  The page must navigate to /Checkout URL.
    } else {
      errorNoticeDomNodeRef.current.innerHTML = `Error.  Cannot proceed to checkout.  ${text}.`
    }
  })
  .catch((err) => {
    //'fetch-error'
    errorNoticeDomNodeRef.current.innerHTML = `Unable to proceed to checkout.  Network error.`;
  });   
}