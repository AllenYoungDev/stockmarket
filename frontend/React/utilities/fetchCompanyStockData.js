'use strict';

import './ServerUrl.js';

export function fetchCompanyStockData(setStockDataServerResponse, ignore) {

    fetch(window.backendServerUrl + "/GetCompanyStockData", {method: "GET", credentials: 'include'})
    .then((response) => response.status === 200 ? response.json() : response.text())
    .then((data) => {
      console.log(`CompanyStockStats() typeof data:  ${typeof data}.`)
      if (typeof data === 'object') {
        console.log(`CompanyStockStats() JSON response:  ${data}.`)
        //console.log(`CompanyStockStats() data[0]:  ${data[0]}.`)
        if (!ignore) {
          setStockDataServerResponse({responseType: 'success', stockData: data});
        }
      } else {
        console.error(`CompanyStockStats() text response:  ${data}.`)
        if (!ignore) {
          setStockDataServerResponse({responseType: 'server-error', stockData: {}});
        }      
      }
    })
    .catch((err) => {
      console.error(`CompanyStockStats() fetch() error:  ${err}.`)
      if (!ignore) {
        setStockDataServerResponse({responseType: 'fetch-error', stockData: {}});
      }    
    });
}