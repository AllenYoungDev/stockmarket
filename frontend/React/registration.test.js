/**
 * @jest-environment jsdom
 */

//import React from "react";
//import ReactDOM from 'react-dom/client';

import renderer from 'react-test-renderer';
import {UserRegistrationPage, AccountRegistrationForm} from './registration.js';

jest.mock('./utilities/fetchAccessTokenValidity');

import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity';

(async () => {
  if (typeof React === 'undefined') {
    globalThis.React = await import("react");
  }
  if (typeof ReactDOM === 'undefined') {
    globalThis.ReactDOM = await import("react-dom/client");
  }
})();

var loginState = '';

async function registrationPageJestSnapshotTest(loginStateArg) {
  let testRenderer;
  let testInstance;
  let tree;

  loginState = loginStateArg;

  //renderer.create() is wrapped in renderer.act() for executing useEffect() calls in <HomePage />.
  //https://github.com/facebook/react/issues/15321
  await renderer.act(async () => {
    testRenderer = renderer.create(
      <UserRegistrationPage />,
    );
  });

  testInstance = testRenderer.root;
  expect(testInstance.findByType(AccountRegistrationForm).props.loginState).toBe(loginStateArg);

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
}

it('React app Jest snapshot testing for Allen Young\'s Stockmarket Demo registration page', async () => {

  fetchAccessTokenValidity.mockImplementation((setLoginState, ignore) => {
    console.log('fetchAccessTokenValidity() mock called!');
    setLoginState(loginState);
  });

  /*************************************************************************************************
  No-login mock Jest snapshot test
  **************************************************************************************************/
  await registrationPageJestSnapshotTest('no-login');

  /*************************************************************************************************
  Non-admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await registrationPageJestSnapshotTest('non-admin-login');

  /*************************************************************************************************
  Admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await registrationPageJestSnapshotTest('admin-login');

  /*************************************************************************************************
  Fetch-error mock Jest snapshot test
  **************************************************************************************************/
  await registrationPageJestSnapshotTest('fetch-error');
});