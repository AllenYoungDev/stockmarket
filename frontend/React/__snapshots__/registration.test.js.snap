// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo registration page 1`] = `
[
  <div
    className="top-bar"
  >
    <a
      href="http://localhost"
      style={
        {
          "color": "rgb(255,255,255)",
          "textDecoration": "none",
        }
      }
    >
      Allen Young's Stockmarket Demo
    </a>
  </div>,
  <section
    className="registration-form-section"
  >
    <h1
      className="registration-form-section-heading"
    >
      Allen Young's Stockmarket Demo
      <br />
      Account Registration
    </h1>
    <form
      className="registration-form"
    >
      <div
        className="registration-form-div-grid"
      >
        <label
          className="registration-form-label-first-name"
          htmlFor="firstname"
        >
          First name
        </label>
        <input
          className="registration-form-input-first-name"
          id="firstname"
          name="firstname"
          placeholder="First name"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-last-name"
          htmlFor="lastname"
        >
          Last name
        </label>
        <input
          className="registration-form-input-last-name"
          id="lastname"
          name="lastname"
          placeholder="Last name"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-first-last-names"
        />
        <label
          className="registration-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="registration-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="registration-form-input-phone-number"
          id="phonenumber"
          name="phonenumber"
          placeholder="Phone number"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-phone-number"
        />
        <label
          className="registration-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="registration-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
        <p
          className="registration-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
        <label
          className="registration-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="registration-form-input-password-retype"
          id="passwordretype"
          name="passwordretype"
          placeholder="Password (Retype)"
          size="20"
          type="password"
        />
      </div>
      <p
        className="registration-form-paragraph-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="registration-form-paragraph-notice2"
      >
        Your email address will be your login ID.
      </p>
      <p
        className="registration-form-paragraph-notice3"
      >
        After pressing the register button below, you'll receive an email-address verification email with a link that you must click to verify your email address.  You will not be able to log in before verifying your email address.
      </p>
      <div
        className="registration-form-div-register-button"
      >
        <p
          className="registration-form-error-notice"
        />
        <input
          className="registration-form-input-register-button"
          onClick={[Function]}
          type="button"
          value="Register"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      Â© 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo registration page 2`] = `
[
  <div
    className="top-bar"
  >
    <a
      href="http://localhost"
      style={
        {
          "color": "rgb(255,255,255)",
          "textDecoration": "none",
        }
      }
    >
      Allen Young's Stockmarket Demo
    </a>
  </div>,
  <section
    className="registration-form-section"
  >
    <h1
      className="registration-form-section-heading"
    >
      Allen Young's Stockmarket Demo
      <br />
      Account Registration
    </h1>
    <form
      className="registration-form"
    >
      <div
        className="registration-form-div-grid"
      >
        <label
          className="registration-form-label-first-name"
          htmlFor="firstname"
        >
          First name
        </label>
        <input
          className="registration-form-input-first-name"
          disabled={true}
          id="firstname"
          name="firstname"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-last-name"
          htmlFor="lastname"
        >
          Last name
        </label>
        <input
          className="registration-form-input-last-name"
          disabled={true}
          id="lastname"
          name="lastname"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-first-last-names"
        />
        <label
          className="registration-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="registration-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="registration-form-input-phone-number"
          disabled={true}
          id="phonenumber"
          name="phonenumber"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-phone-number"
        />
        <label
          className="registration-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="registration-form-input-password"
          disabled={true}
          id="password"
          name="password"
          type="text"
        />
        <label
          className="registration-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="registration-form-input-password-retype"
          disabled={true}
          id="passwordretype"
          name="passwordretype"
          size="20"
          type="text"
        />
      </div>
      <p
        className="registration-form-paragraph-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="registration-form-paragraph-notice2"
      >
        Your email address will be your login ID.
      </p>
      <p
        className="registration-form-paragraph-notice3"
      >
        After pressing the register button below, you'll receive an email-address verification email with a link that you must click to verify your email address.  You will not be able to log in before verifying your email address.
      </p>
      <div
        className="registration-form-div-register-button"
      >
        <p
          className="registration-form-error-notice"
        />
        <input
          className="registration-form-input-register-button"
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Register"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      Â© 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo registration page 3`] = `
[
  <div
    className="top-bar"
  >
    <a
      href="http://localhost"
      style={
        {
          "color": "rgb(255,255,255)",
          "textDecoration": "none",
        }
      }
    >
      Allen Young's Stockmarket Demo
    </a>
  </div>,
  <section
    className="registration-form-section"
  >
    <h1
      className="registration-form-section-heading"
    >
      Allen Young's Stockmarket Demo
      <br />
      Account Registration
    </h1>
    <form
      className="registration-form"
    >
      <div
        className="registration-form-div-grid"
      >
        <label
          className="registration-form-label-first-name"
          htmlFor="firstname"
        >
          First name
        </label>
        <input
          className="registration-form-input-first-name"
          disabled={true}
          id="firstname"
          name="firstname"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-last-name"
          htmlFor="lastname"
        >
          Last name
        </label>
        <input
          className="registration-form-input-last-name"
          disabled={true}
          id="lastname"
          name="lastname"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-first-last-names"
        />
        <label
          className="registration-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="registration-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="registration-form-input-phone-number"
          disabled={true}
          id="phonenumber"
          name="phonenumber"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-phone-number"
        />
        <label
          className="registration-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="registration-form-input-password"
          disabled={true}
          id="password"
          name="password"
          type="text"
        />
        <label
          className="registration-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="registration-form-input-password-retype"
          disabled={true}
          id="passwordretype"
          name="passwordretype"
          size="20"
          type="text"
        />
      </div>
      <p
        className="registration-form-paragraph-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="registration-form-paragraph-notice2"
      >
        Your email address will be your login ID.
      </p>
      <p
        className="registration-form-paragraph-notice3"
      >
        After pressing the register button below, you'll receive an email-address verification email with a link that you must click to verify your email address.  You will not be able to log in before verifying your email address.
      </p>
      <div
        className="registration-form-div-register-button"
      >
        <p
          className="registration-form-error-notice"
        />
        <input
          className="registration-form-input-register-button"
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Register"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      Â© 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo registration page 4`] = `
[
  <div
    className="top-bar"
  >
    <a
      href="http://localhost"
      style={
        {
          "color": "rgb(255,255,255)",
          "textDecoration": "none",
        }
      }
    >
      Allen Young's Stockmarket Demo
    </a>
  </div>,
  <section
    className="registration-form-section"
  >
    <h1
      className="registration-form-section-heading"
    >
      Allen Young's Stockmarket Demo
      <br />
      Account Registration
    </h1>
    <form
      className="registration-form"
    >
      <div
        className="registration-form-div-grid"
      >
        <label
          className="registration-form-label-first-name"
          htmlFor="firstname"
        >
          First name
        </label>
        <input
          className="registration-form-input-first-name"
          id="firstname"
          name="firstname"
          placeholder="First name"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-last-name"
          htmlFor="lastname"
        >
          Last name
        </label>
        <input
          className="registration-form-input-last-name"
          id="lastname"
          name="lastname"
          placeholder="Last name"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-first-last-names"
        />
        <label
          className="registration-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="registration-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="registration-form-label-phone-number"
          htmlFor="phonenumber"
        >
          Phone number
        </label>
        <input
          className="registration-form-input-phone-number"
          id="phonenumber"
          name="phonenumber"
          placeholder="Phone number"
          size="20"
          type="text"
        />
        <div
          className="registration-form-div-below-phone-number"
        />
        <label
          className="registration-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="registration-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          type="password"
        />
        <p
          className="registration-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
        <label
          className="registration-form-label-password-retype"
          htmlFor="passwordretype"
        >
          Password (Retype)
        </label>
        <input
          className="registration-form-input-password-retype"
          id="passwordretype"
          name="passwordretype"
          placeholder="Password (Retype)"
          size="20"
          type="password"
        />
      </div>
      <p
        className="registration-form-paragraph-notice1"
      >
        Password must be at least 10 characters, and must contain a letter, a number, and a special character from any of the following ~\`!@#$%^&*-_+=|\\/?;:"'&lt;&gt;,.
        []().
      </p>
      <p
        className="registration-form-paragraph-notice2"
      >
        Your email address will be your login ID.
      </p>
      <p
        className="registration-form-paragraph-notice3"
      >
        After pressing the register button below, you'll receive an email-address verification email with a link that you must click to verify your email address.  You will not be able to log in before verifying your email address.
      </p>
      <div
        className="registration-form-div-register-button"
      >
        <p
          className="registration-form-error-notice"
        />
        <input
          className="registration-form-input-register-button"
          onClick={[Function]}
          type="button"
          value="Register"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      Â© 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;
